packages/wallet/src/utils/json-rpc-utils.ts

```diff
@@ -3,68 +3,36 @@ import {
   State,
   Channel,
   isAllocationOutcome,
-  SignedState,
   convertBytes32ToAddress,
   convertAddressToBytes32
 } from "@statechannels/nitro-protocol";
 import {bigNumberify, randomBytes} from "ethers/utils";
 import {NETWORK_ID, CHALLENGE_DURATION, ETH_ASSET_HOLDER_ADDRESS} from "../constants";
-import {ChannelParticipant} from "../redux/channel-store";
-import {RelayableAction} from "../communication";
-
-export interface JsonRpcParticipant {
-  participantId: string;
-  signingAddress: string;
-  destination: string;
-}
-
-export interface JsonRpcAllocationItem {
-  destination: string;
-  amount: string;
-}
-export type JsonRpcAllocations = JsonRpcAllocation[];
-
-export interface JsonRpcAllocation {
-  token: string;
-  allocationItems: JsonRpcAllocationItem[];
-}
-
-export interface JsonRpcCreateChannelParams {
-  participants: JsonRpcParticipant[];
-  allocations: JsonRpcAllocations;
-  appDefinition: string;
-  appData: string;
-}
-
-export interface JsonRpcMessage {
-  recipient: string;
-  sender: string;
-  data: WalletMessage;
-}
-
-interface OpenChannel {
-  type: "Channel.Open";
-  participants: ChannelParticipant[];
-  signedState: SignedState;
-}
-interface ChannelJoined {
-  type: "Channel.Joined";
-  signedState: SignedState;
-}
-
-interface ChannelUpdated {
-  type: "Channel.Updated";
-  signedState: SignedState;
-}
-type WalletMessage = OpenChannel | ChannelJoined | RelayableAction | ChannelUpdated;
-
-export interface JsonRpcUpdateChannelParams {
-  allocations: JsonRpcAllocations;
-  appData: string;
-  channelId: string;
-}
-
-function createAllocationOutcomeFromParams(params: JsonRpcAllocations): Outcome {
+import {
+  UpdateChannelParams,
+  Allocations,
+  CreateChannelParams
+} from "@statechannels/client-api-schema";
+
+// TODO: Figure out how these should have been used
```

Comment:
@lalexgap

Reply:
These should probably be moved somewhere else. These are interfaces for messages between `wallet`s.

The json-rpc api specifies that the `data` on a message can be anything. These interfaces define what the `wallet` currently expects/handles in the `data` field.

```
{
  "jsonrpc": "2.0",
  "method": "PushMessage",
  "id": 1,
  "params": {
    "recipient": "user123",
    "sender": "user456",
    "data": "0x123.."
  }
}
```